module-4/backend/week_one.md

```diff
@@ -1,19 +1,42 @@
 ## Week One - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
 
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
 1. What's the most useful thing you learned from completing the intermission week work?
+The most useful thing I learned from completing the intermission week work was going through the JavaScript fundamentals. It was useful to go over the material, run through the lessons exercises, and take notes. I didn't have the opportunity to get to any of the extensions but I think going through additional exercises like the homework from the first week would have me feeling more confident in JavaScript. I had even thought about doing front-end Mythical Creatures.
+
 2. What are some tools to help debug JavaScript code?
+Some tools I was previously utilizing to help debug JavaScript code include the use of `console.log()` to see what return values I get from functions and variables, `eval(pry.it)` (also, `debugger;`). After Friday's lesson I found that Chrome's developer tools (Element, Console, Source and Network tabs) are also very useful in assisting in JavaScript debugging and expanding my understanding of the performance of my applications.
+
 3. What are some tools you need in order to unit test your JavaScript?
+Some tools needed in order to unit test JavaScript are Mocha and Chai. Mocha is used for running tests and provieds `describe()`, `context()` and `it()` functionality, while Chai is a popular assertion library.
+
 4. What is the syntax for invoking a function? Give an example.
+You may have a function called `adoptCat` which does x, y and z. Simply calling `adoptCat` in the console will return that it is a function, but that function will not be invoked unless it is called like so: `adoptCat()`
+
 5. What's `this` in JavaScript?
+JavaScript's `this` refers to the parent object, and is dependent on the context in which it was referenced. When used in the global context, `this` refers to 'document' or 'window'. In the context of an object, `this` refers to and is boud to the object itself.
+
 6. What is Webpack and why is it useful?
+Webpack is a Node package that inspects our asses, finds dependencies, and creates a JS file (main.bundle.js) that is used for production.
```

Comment:
Yup! We can also add transpilers such as babel to convert ES6 to ES5. It helps us build our code